core/store/src/lib.rs

```diff
@@ -522,6 +522,12 @@ impl StoreUpdate {
         self.transaction.delete_all(column);
     }
 
+    /// Deletes the given key range from the database including `from`
+    /// and excluding `to` keys.
+    pub fn delete_range(&mut self, column: DBCol, from: &[u8], to: &[u8]) {
```

Comment:
I know this is a pre-existing issue with the sibling methods, but it is generally a good idea to hoist the conversions like `to_vec` to the caller. That way if they do happen to already happen to have a `Vec<u8>` on hand they’re willing to give away, a full copy of the contents can be avoided.

Reply:
Yeah, that makes sense. I've looked into refactoring `StoreUpdate` methods to accept `Vec<u8>` instead of `&[u8]`, but that requires changes in many places, not sure if it is worth it.